--- reglookup-src-1.0.0/SConstruct~	2011-06-13 04:27:42.000000000 +0200
+++ reglookup-src-1.0.0/SConstruct	2011-06-20 09:48:52.902314648 +0200
@@ -44,18 +44,33 @@
 
 # Installation
 prefix='/usr/local/'
+destdir=''
+bindir= prefix + 'bin'
+libdir=prefix + 'lib'
+includedir=prefix + 'include'
+mandir=prefix + 'man'
 if 'PREFIX' in os.environ:
    prefix = os.environ['PREFIX']+'/'
-
-install_items = [prefix+'bin',
-                 prefix+'lib', 
-                 prefix+'include/regfi',
-                 prefix+'man']
-
-env.Install(prefix+'bin', [reglookup, reglookup_recover, 'bin/reglookup-timeline'])
-env.Install(prefix+'lib', [libregfi, libregfi_static])
-env.Install(prefix+'include/regfi', Glob('include/*.h'))
-env.Install(prefix+'man/man1', [man_reglookup, man_reglookup_recover,
+if 'DESTDIR' in os.environ:
+   destdir = os.environ['DESTDIR']
+if 'BINDIR' in os.environ:
+   bindir = os.environ['BINDIR']
+if 'LIBDIR' in os.environ:
+   libdir = os.environ['LIBDIR']
+if 'INCLUDEDIR' in os.environ:
+   includedir = os.environ['INCLUDEDIR']
+if 'MANDIR' in os.environ:
+   mandir = os.environ['MANDIR']
+
+install_items = [destdir + bindir,
+                 destdir + libdir, 
+                 destdir + includedir + '/regfi',
+                 destdir + mandir + '/man1']
+
+env.Install(destdir + bindir, [reglookup, reglookup_recover, 'bin/reglookup-timeline'])
+env.Install(destdir + libdir, [libregfi, libregfi_static])
+env.Install(destdir + includedir + '/regfi', Glob('include/*.h'))
+env.Install(destdir + mandir + '/man1', [man_reglookup, man_reglookup_recover,
                                 man_reglookup_timeline])
 
 if sys.version_info[0] == 2:
@@ -63,7 +76,7 @@
    env.Command('pyregfi2-install.log', ['python/pyregfi/__init__.py', 
                                         'python/pyregfi/structures.py', 
                                         'python/pyregfi/winsec.py'],
-               "python pyregfi-distutils.py install | tee pyregfi2-install.log")
+               "python pyregfi-distutils.py install --root=" + destdir + " | tee pyregfi2-install.log")
 
 python_path = os.popen('which python3').read()
 if python_path != '':
@@ -71,7 +84,7 @@
    env.Command('pyregfi3-install.log', ['python/pyregfi/__init__.py', 
                                         'python/pyregfi/structures.py', 
                                         'python/pyregfi/winsec.py'], 
-               "python3 pyregfi-distutils.py install | tee pyregfi3-install.log")
+               "python3 pyregfi-distutils.py install --root=" + destdir + " | tee pyregfi3-install.log")
 
 # API documentation
 regfi_doc = env.Command('doc/devel/regfi/index.html', 
